server/src/main/java/keywhiz/service/permissions/PermissionCheckModule.java

```diff
@@ -8,13 +8,31 @@
 public class PermissionCheckModule extends AbstractModule {
 
   @Override
-  protected void configure() {}
+  protected void configure() {
+    bindToDefaultConstructor(AllowClientSecretCreationPermissionCheck.class);
+  }
```

Comment:
Elsewhere in the code we have a "strict Guice module" which disables automatically injecting classes with a no-argument constructor, which is a slightly error-prone default behavior of Guice (the bugs this can introduce are rare but very confusing). This includes synthetic (i.e. generated by the compiler) constructors, so even though our "allow clients to create secrets" permission check doesn't appear to have a constructor at all, it still gets caught up.

So, rather than create an empty constructor just to slap @ Inject on it, I decided to write a little utility method and bind the default constructor directly instead, which also satisfies the strict requirement.

Reply:
So when AllowClientSecretCreationPermissionCheck is used in for PermissionCheck provider method, it's created following this constructor?

Reply:
Yup! We're just telling Guice that when it needs an instance of this class, it should use this constructor.